code/app/src/androidTest/java/com/example/bread/MoodHistoryFragmentTest.java

```diff
@@ -0,0 +1,163 @@
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.FirebaseFirestore;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class MoodHistoryFragmentTest {
+
+    @BeforeClass
+    public static void setup(){
+        // Specific address for emulated device to access our localHost
+        String androidLocalhost = "10.0.2.2";
+        int portNumber = 8080;
+        FirebaseFirestore.getInstance().useEmulator(androidLocalhost, portNumber);
+    }
+
+    @Rule
+    public ActivityScenarioRule<HomePage> activityRule = new ActivityScenarioRule<>(HomePage.class);
+
+    //https://firebase.google.com/docs/firestore/manage-data/add-data
+    @Before
+    public void seedDatabase() throws ExecutionException, InterruptedException {
+        //creating firebase instance
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+        Map<String, Object> user = new HashMap<>();
+
+        user.put("username", "testUser");
+        DocumentReference participantRef = db.collection("participants").document("testUser");
+        participantRef.set(user);
+
+        Tasks.await(participantRef.set(user));
+
+        MoodEvent mood1 = new MoodEvent("mood1", "so tired", MoodEvent.EmotionalState.SAD, participantRef);
+        MoodEvent mood2 = new MoodEvent("mood2", "coffee time", MoodEvent.EmotionalState.HAPPY, participantRef);
+        MoodEvent mood3 = new MoodEvent("mood3", "exam time", MoodEvent.EmotionalState.ANXIOUS, participantRef);
+        MoodEvent mood4 = new MoodEvent("mood4", "failed it", MoodEvent.EmotionalState.SHAMEFUL, participantRef);
+
+        mood1.setTimestamp(new GregorianCalendar(2025, Calendar.MARCH, 1).getTime());
+        mood2.setTimestamp(new GregorianCalendar(2025, Calendar.MARCH, 2).getTime());
+        mood3.setTimestamp(new GregorianCalendar(2025, Calendar.MARCH, 3).getTime());
+        mood4.setTimestamp(new GregorianCalendar(2025, Calendar.MARCH, 4).getTime());
+
+        db.collection("moodEvents").document("mood1").set(mood1);
+        db.collection("moodEvents").document("mood2").set(mood2);
+        db.collection("moodEvents").document("mood3").set(mood3);
+        db.collection("moodEvents").document("mood4").set(mood4);
+
+        DocumentReference mood1Ref = db.collection("moodEvents").document("mood1");
+        Tasks.await(mood1Ref.set(mood1));
+        }
+
+    @Test
+    public void displaysAllMoodEventsTest() throws InterruptedException, ExecutionException {
+        seedDatabase();
```

Comment:
You don't need to call seedDatabase again here. `@Before` calls it automatically before every test